kubernetes/resource_kubernetes_annotations.go

```diff
@@ -184,6 +215,32 @@ func getManagedAnnotations(managedFields []v1.ManagedFieldsEntry, manager string
 	return annotations, nil
 }
 
+// getTemplateManagedAnnotations reads the field manager metadata to discover which fields we're managing
+func getTemplateManagedAnnotations(managedFields []v1.ManagedFieldsEntry, manager string, kind string) (map[string]interface{}, error) {
+	var annotations map[string]interface{}
+	for _, m := range managedFields {
+		if m.Manager != manager {
+			continue
+		}
+		var mm map[string]interface{}
+		err := json.Unmarshal(m.FieldsV1.Raw, &mm)
+		if err != nil {
+			return nil, err
+		}
+		spec := mm["f:spec"].(map[string]interface{})
```

Comment:
If the Terraform code does not have `template_annotations` but only `annotations` the provider will panic here. That is because `mm` contains only `metadata.annotations` but not `spec.template.metadata.annotations` as expected.

I cannot comment on function `getManagedAnnotations` but similar behavior when the Terraform code does not have `annotations` but only `template_annotations`. `mm` there will be `spec.template.metadata.annotations`  but not `metadata. annotations` as expected and thus will panic as well.

One more edge case scenario is when we have both `template_annotations` and `annotations` and then for whatever reason, the managed annotations or template annotations of the Kubernetes object were updated by someone else, i.e. `Terraform` field manager is not there anymore, we will get one of the above conditions as well.